user_tools/src/spark_rapids_pytools/rapids/qualification.py

```diff
@@ -572,9 +572,9 @@ def __calc_apps_cost(self,
         def get_costs_for_single_app(df_row, estimator: SavingsEstimator) -> pd.Series:
             raw_cpu_cost, raw_gpu_cost, _ = estimator.get_costs_and_savings(df_row['App Duration'],
                                                                             df_row['Estimated GPU Duration'])
-            cpu_cost = (100 - self.ctxt.get_ctxt('cpu_discount')) / 100 * raw_cpu_cost
-            gpu_cost = (100 - self.ctxt.get_ctxt('gpu_discount')) / 100 * raw_gpu_cost
-            est_savings = 100.0 - ((100.0 * gpu_cost) / cpu_cost)
+            cpu_cost = (100 - self.ctxt.get_ctxt('cpu_discount')) / 100 * raw_cpu_cost if raw_cpu_cost > 0 else 0
+            gpu_cost = (100 - self.ctxt.get_ctxt('gpu_discount')) / 100 * raw_gpu_cost if raw_gpu_cost > 0 else 0
```

Comment:
I think we don't need to change `cpu_cost` or `gpu_cost` because the expression will perform division `/100` first then do the multiplication `*raw_cpu/gpu_cost` so the raw costs can be 0 if they are.

Reply:
yes, corrected. I think I force pushed, sorry about that.